watcher/watcher.go

```diff
@@ -600,75 +594,206 @@ func (w *Watcher) checkState(
 	roomCount *models.RoomsStatusCount,
 	scheduler *models.Scheduler,
 	nowTimestamp int64,
-) (bool, bool, bool, error) { //shouldScaleUp, shouldScaleDown, changedState
-	var shouldScaleUp, shouldScaleDown, changedState bool
-	inSync := true
+) (*scaling, error) { //delta, shouldScaleUp, shouldScaleDown, changedState
+	var err error
+	scaling := &scaling{
+		Delta:        0,
+		ChangedState: false,
+		InSync:       true,
+	}
 
+	// Creating or Terminating state
 	if scheduler.State == models.StateCreating || scheduler.State == models.StateTerminating {
-		return false, false, changedState, nil
+		return scaling, nil
 	}
+	// Rooms below Min
 	if roomCount.Total() < autoScalingInfo.Min {
-		return true, false, changedState, nil
+		scaling.Delta = autoScalingInfo.Min - roomCount.Total()
+		return scaling, nil
 	}
-	if autoScalingInfo.Min > 0 && 100*roomCount.Occupied >= roomCount.Total()*autoScalingInfo.Up.Trigger.Limit {
-		return true, false, changedState, nil
+	// Rooms above Max
+	if autoScalingInfo.Max > 0 && roomCount.Total() > autoScalingInfo.Max {
+		scaling.Delta = autoScalingInfo.Max - roomCount.Total()
+		return scaling, nil
 	}
 
-	threshold := autoScalingInfo.Up.Trigger.Threshold
-	usage := float32(autoScalingInfo.Up.Trigger.Usage) / 100
-	capacity := autoScalingInfo.Up.Trigger.Time / w.AutoScalingPeriod
+	// Send current usage
+	if len(autoScalingInfo.Down.MetricsTrigger) > 0 {
```

Comment:
This method is too long. You could separate the ifs into methods whose names are their intention.